api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2ClientConfig.java

```diff
@@ -200,30 +219,64 @@ private static HttpClientException genericError(Status status) {
         "OAuth2 server replied with HTTP status code: " + status.getCode());
   }
 
+  private static void check(
+      List<String> violations, String paramKey, boolean cond, String msg, Object... args) {
+    if (!cond) {
+      if (args.length == 0) {
+        violations.add(msg + " (" + paramKey + ")");
+      } else {
+        violations.add(format(msg, args) + " (" + paramKey + ")");
+      }
+    }
+  }
+
   @Value.Check
   void check() {
-    checkArgument(
+    List<String> violations = new ArrayList<>();
+
+    check(
+        violations,
+        CONF_NESSIE_OAUTH2_CLIENT_ID,
+        !getClientId().isEmpty(),
+        "client ID must not be empty");
+    check(
+        violations,
+        CONF_NESSIE_OAUTH2_CLIENT_SECRET,
+        getClientSecret().length() > 0,
```

Comment:
nit: why do we do `!getClientId().isEmpty()` above but check length here?

Reply:
it's `Secret`